src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/Codes/Codes.tsx

```diff
@@ -119,23 +150,59 @@ const Codes = ({
     setSavedCode(cqlCode);
   };
 
+  // Checks if new code is a duplicate based on chip ids.
   const addNewCode = () => {
     if (selectedCodeSystemName === "Custom") {
-      const customCqlCode = new CQL.Code(
-        customCodeConcept,
-        customCodeSystemName, // What is the oid for custom CS
-        null,
-        customCodeConcept
-      );
-      handleChange(customCqlCode);
+      const newCodeId = `${customCodeSystemName}-${customCodeConcept}`;
+      const existingCode = _.filter(chips, _.matches({ id: newCodeId }));
```

Comment:
if you just want to check if an element is present, `find` or `some` is fine I guess